### PR TITLE
Fixes #1944 - Create pagination component

### LINKED ISSUE #1944: Migrate clay-pagination and clay-pagination-bar to React
Look into combining these components but they may still need to be separate.

### PR DESCRIPTION
**Pull request by bryceosterhaus**

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: packages/clay-pagination/src/PaginationEllipsis.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}) => {
+	const [active, setActive] = React.useState(false);
+
+	return (
+		<ClayDropDown
+			active={active}
+			className="page-item"
+			containerElement="li"
+			onActiveChange={newVal => setActive(newVal)}
+			trigger={<ClayButton className="page-link">{'...'}</ClayButton>}
```
Comment (matuzalemsteles) [inline packages/clay-pagination/src/PaginationEllipsis.tsx]: hey @bryceosterhaus maybe we'll just have to use the `button` tag, we'll have problems with the `btn-*`class that are added by default to the ClayButton.

File: packages/clay-pagination/src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+			Math.max(activeIndex - ellipsisBuffer, 1),
+			// Add 1 to account for active index
+			Math.min(activeIndex + ellipsisBuffer + 1, lastIndex)
+		),
+		...rightBuffer,
+		pages[lastIndex],
+	];
+};
+
+interface Props extends React.HTMLAttributes<HTMLUListElement> {
```
Comment (matuzalemsteles) [inline packages/clay-pagination/src/index.tsx]: Remember to add documentation to this interface.

File: packages/clay-pagination/src/PaginationEllipsis.tsx (hunk 1 above)
Comment (bryceosterhaus) [reply packages/clay-pagination/src/PaginationEllipsis.tsx]: The clay docs show you can use clay-button with pagination. Check out, https://clayui.com/docs/css-framework/satellites/paginations.html. I forgot to add the `btn-unstyled` though, I will update that.

**Comment by bryceosterhaus** [comment]
@matuzalemsteles updated

File: packages/clay-drop-down/src/index.tsx [hunk 3]
```diff
@@ -72,11 +79,19 @@ const ClayDropDown: React.FunctionComponent<Props> & {
 	const triggerElementRef = React.useRef<HTMLButtonElement>(null);
 	const menuElementRef = React.useRef<HTMLDivElement>(null);
 
+	const ContainerElement = containerElement;
```
Comment (matuzalemsteles) [inline packages/clay-drop-down/src/index.tsx]: Just a Tip, you can define an alias for `containerElement` without having to declare a variable here.

```
containerElement: ContainerElement = 'div',
```

**Comment by bryceosterhaus** [comment]
@matuzalemsteles went ahead and added the alias you mentioned.

**Comment by matuzalemsteles** [comment]
Thanks @bryceosterhaus

**Pull request merged**